src/perform_step/firk_perform_step.jl

```diff
@@ -264,7 +264,8 @@ end
 
     @. dw12 = complex(fw1 - αdt*Mw1 + βdt*Mw2, fw2 - βdt*Mw1 - αdt*Mw2)
     needfactor = iter==1
-    linsolve2(vec(dw12), W1, vec(dw12), needfactor)
+    linsolve2(vec(cubuff), W1, vec(dw12), needfactor)
+    @.. dw12 = cubuff
```

Comment:
You don't really need to copy the data. Just do a pointer reassignment is enough
```julia
dw12 = cubuff
```
and since `cubuff` and `dw12` will be unpacked again from the cache, no information is lost.

Reply:
I just feared that the reuse of `cubuff`/`ubuff`could cause some trouble, but you are right.

Reply:
I have to take back what I said. 

> and since cubuff and dw12 will be unpacked again from the cache, no information is lost.
this peace of code is running in a loop, so it will brake again if more then one iteration is happens.